packages/flutter/lib/src/services/restoration.dart

```diff
@@ -294,23 +294,53 @@ class RestorationManager extends ChangeNotifier {
     _bucketsNeedingSerialization.remove(bucket);
   }
 
-  void _doSerialization() {
+  Future<void> _doSerialization() {
+    if (!_serializationScheduled) {
+      return Future<void>.value(null);
+    }
     assert(() {
       _debugDoingUpdate = true;
       return true;
     }());
-    _postFrameScheduled = false;
+    _serializationScheduled = false;
 
     for (final RestorationBucket bucket in _bucketsNeedingSerialization) {
       bucket.finalize();
     }
     _bucketsNeedingSerialization.clear();
-    sendToEngine(_encodeRestorationData(_rootBucket!._rawData));
+    final Future<void> result = sendToEngine(_encodeRestorationData(_rootBucket!._rawData));
 
     assert(() {
       _debugDoingUpdate = false;
       return true;
     }());
+    return result;
+  }
+
+  /// Called to manually flush the restoration data to the engine.
+  ///
+  /// Restoration data is automatically flushed to the engine at the end of a
+  /// frame. Since a change in restoration data is usually accompanied by
+  /// scheduling a frame (e.g. because the restoration data is modified inside a
+  /// [State.setState] call) it is uncommon to call this method directly.
+  /// However if restoration data is changed without triggering a frame this
+  /// method must be called to ensure that the updated restoration data is send
+  /// to the engine in a timely manner. An example for such a use case is the
+  /// [Scrollable], where the final to be persisted scroll offset after a scroll
```

Comment:
"final to be persisted scroll offset" is dubious, not sure what you mean

Reply:
Reworded.